### PR TITLE
add CI scripts and templates

### PR DESCRIPTION
**Pull request by NautiluX**

### CODE REVIEW

File: hack/generate_template.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    # for each subdir of yaml_directory append 'object' to template
+    for (dirpath, dirnames, filenames) in os.walk(arguments.yaml_directory):
+        # skip 'crds' directory, these are bundled in the CSV
+        if filenames:
+            sss_name = dirpath.replace('/','-').replace(arguments.yaml_directory.replace('/','-'),arguments.repo_name)
+            process_yamls(sss_name, dirpath, selectorsyncset_data)
+
+    # write template file ordering by keys
+    with open(arguments.destination,'w') as outfile:
+        yaml.dump(template_data, outfile, default_flow_style=False)
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply hack/generate_template.py:123]: it seems to be used _somewhere_ so I copied it over as is from RMO

File: hack/pipeline.dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS kustomize-builder
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply hack/pipeline.dockerfile]: that's right... let me fix that

[the remark this answers is not in the document]
Comment (NautiluX) [reply hack/pipeline.dockerfile]: oh it's not using this image anymore in the latest commit anyways.

**Comment by mrWinston** [comment]
Ran the following instructions:
```
export REGISTRY_TOKEN=...
export REGISTRY_USER=mdewald
export QUAY_USER=$REGISTRY_USER
export QUAY_TOKEN=$REGISTRY_TOKEN
export IMG=quay.io/<username>/managed-node-metadata-operator
export IMAGE_REPOSITORY=$REGISTRY_USER
export QUAY_IMAGE=$IMG
export CONTAINER_ENGINE=docker 
docker login quay.io
./hack/app_sre_build_deploy.sh
```
and the build worked out fine.

**Pull request merged**